drivers/storage/portworx/deployment.go

```diff
@@ -1295,9 +1295,39 @@ func (t *template) getVolumeMounts() []v1.VolumeMount {
 	for _, fn := range extensions {
 		volumeInfoList = append(volumeInfoList, fn()...)
 	}
+	volumeInfoList = t.overrideVolumeMount(volumeInfoList)
 	return t.mountsFromVolInfo(volumeInfoList)
 }
 
+func (t *template) overrideVolumeMount(volInfos []volumeInfo) []volumeInfo {
+	volPaths := map[string]int{}
+
+	for i := range volInfos {
+		volPaths[volInfos[i].mountPath] = i
+	}
+
+	for i := range t.cluster.Spec.Volumes {
+		v := t.cluster.Spec.Volumes[i]
+		existingVolIdx, ok := volPaths[v.MountPath]
+		if !ok {
+			volInfos = append(volInfos, volumeInfo{
+				name:             v.Name,
+				readOnly:         v.ReadOnly,
+				hostPath:         v.HostPath.String(),
+				mountPath:        v.MountPath,
+				mountPropagation: v.MountPropagation,
+				hostPathType:     v.HostPath.Type,
+				configMapType:    v.ConfigMap,
+			})
+		} else {
+			volInfos[existingVolIdx].hostPath = v.HostPath.String()
+			volInfos[existingVolIdx].mountPropagation = v.MountPropagation
```

Comment:
Why are we overriding _only_ the host-path and the mount-propagation?

Shouldn't we at least copy the mount-options (i.e. read-only)?

Reply:
for existing mounts, if a stc yaml overrides the mount options then it could affect the functionality of porx
imagine someone makes the oci log file to be read only then we do not get to write the logs